### PR TITLE
Only register one gossip validator for full nodes

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Comment by andresilva** [comment]
@rphmeier Can you confirm this? We want full nodes that aren't validators to also gossip these messages right?

**Comment by bkchr** [comment]
(FWIW I tested this with a local test net with 2 validators and one full node)

File: service/src/lib.rs [hunk 1]
```diff
@@ -269,8 +269,6 @@ pub fn new_full(config: Configuration<CustomConfiguration, GenesisConfig>)
 			dht_event_rx,
 		);
 		service.spawn_task(authority_discovery);
-	} else {
-		network_gossip::register_non_authority_validator(service.network());
```
Comment (rphmeier) [inline service/src/lib.rs]: cc @andresilva there was some downside that required this, wasn't there?

**Comment by andresilva** [comment]
@rphmeier https://github.com/paritytech/polkadot/pull/487/files#diff-32562a9cc266a6c93662b535a7803970L201 It is registered unconditionally (that was the original issue for this PR, two validators would be registered if not running as a validator).

**Comment by andresilva** [comment]
I think originally we added the `register_non_authority_validator` because the gossip validator was only being registered for authorities, that seems to have changed with the ICMP PR (#304).

**Pull request merged**